11_notify/willshen8/pkg/puppy/lostpuppy.go

```diff
@@ -0,0 +1,35 @@
+package puppy
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/render"
+)
+
+// Payload struct is the format that lost puppy server expects to see
+type Payload struct {
+	ID int `json:"id"`
+}
+
+// HandleLostPuppy processes the payload and determine the response based on id field
+func HandleLostPuppy(w http.ResponseWriter, r *http.Request) {
+	time.Sleep(2 * time.Second)
+	var jsonData Payload
+	if decodeErr := render.DecodeJSON(r.Body, &jsonData); decodeErr != nil {
+		w.WriteHeader(http.StatusUnprocessableEntity)
+		http.Error(w, http.StatusText(http.StatusUnprocessableEntity), http.StatusUnprocessableEntity)
+		return
+	}
+	if jsonData.ID%2 == 0 {
```

Comment:
nit: I would use `if jsonData%2 == 1` - get use to handling error paths inside `if` statements, and handling the normal flow at the left edge of the function: https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

In this case, it is very trivial, but I think it is still good practice to stick to that.

Reply:
Yes, I agree. I have amended it